### ISSUE TITLE
[HOLD for payment 2023-08-07] [$1000] The #announce is hidden for a long workspace name (works fine with small workspace name)

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#19359]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:

1. Go to staging dot on web chrome
2. Create a new workspace and change the name of the workspace to a longer name
3. Go to announce of the workspace and send any message
4. Now click on reply thread and send any message
5. A new page opens up with the workspace header. Click on it
6. Click on share code
7. Notice that the #announce is hidden and is not shown.
8. If you follow the same steps on small workspace name, the #announce is shown. Since #announce is an important identifier, it should not be hidden even if the workspace name is longer

## Expected Result:
The #announce should not be hidden for a long workspace name since it's the name of the room
## Actual Result:
The #announce is hidden for a long workspace name (works fine with small workspace name)
## Workaround:
Can the user still use Expensify without this being fixed? Have you informed them of the workaround?

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [x] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.3.16-6
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/43996225/ebeb5149-4c1b-497f-b8dc-5aef6cd3be28

<img width="960" alt="Untitled" src="https://github.com/Expensify/App/assets/43996225/917061e5-3d88-43a9-85de-17e13022f87e">

**Expensify/Expensify Issue URL:**
**Issue reported by:** @priya-zha
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1684383597774089

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01a124a19ae3e3c33f</li>
        <li>Upwork Job ID: 1662189223091150848</li>
        2023-07-19<li>Automatic offers: </li>
<ul>
<li> |  | 0</li>
<li>priya-zha | Reporter | 25745668</li>
</ul></li>
    </ul>
</details>

**Comment by PrashantMangukiya** [comment]
## Proposal

Posting proposal early as per new [guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#propose-a-solution-for-the-job)

### Please re-state the problem that we are trying to solve in this issue.
The #announce is hidden for a long workspace name (works fine with small workspace name)

### What is the root cause of that problem?
Workspace name rendered as subtitle via code below:
https://github.com/Expensify/App/blob/737862677811acc2b05c68209a56ecd58bee59f3/src/components/QRShare/index.js#L80-L90

We restrict it with ` numberOfLines={1} `  at line 84 so long text will truncate. This is the root cause of the problem.

### What changes do you think we should make in order to solve the problem?
Within `src/components/QRShare/index.js` file we have to update subtitle as shown below. 
i.e. Remove number of lines prop, and make text as center with break all, so if long text comes without space then it can break into multiple lines.

```javascript
{this.props.subtitle && (
    <Text
        ...
        // numberOfLines={1} // *** Remove this ***
        // style={{marginTop: 4}} // *** Old Code ***
        style={[styles.mt1, styles.textAlignCenter, styles.breakAll]} // *** Updated Code ***
       ...
    >
       ...
    </Text>
)}
``` 

This will solve the issue as shown in result.

### What alternative solutions did you explore? (Optional)
None

<details>
<summary>Result</summary>

<img width="374" alt="Web" src="https://github.com/Expensify/App/assets/7823358/1492288e-cfc0-4a68-b98a-6c9df3314dbf">

</details>

**Comment by dukenv0307** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
The #announce is hidden for a long workspace name (works fine with small workspace name)

### What is the root cause of that problem?

https://github.com/Expensify/App/blob/8180113819532fe5503bc3719ba46ffc566e4ead/src/components/QRShare/index.js#L80-L90
We only display subTitles in one line. So the long workspace name will be truncated
### What changes do you think we should make in order to solve the problem?
We can move the workspace name to the beginning of subTitle and still truncate subTitle. 

https://github.com/Expensify/App/blob/8180113819532fe5503bc3719ba46ffc566e4ead/src/libs/ReportUtils.js#L488-L508
We add one more params in getChatRoomSubtitle function called shouldDisplayRoomNameFirst (default is false)
Then updating 
https://github.com/Expensify/App/blob/8180113819532fe5503bc3719ba46ffc566e4ead/src/libs/ReportUtils.js#L507
to
```
return shouldDisplayRoomNameFirst ? [roomName, workspaceName].join(' • ') : [workspaceName, roomName].join(' • ');
```

and pass shouldDisplayRoomNameFirst is true in here
https://github.com/Expensify/App/blob/8180113819532fe5503bc3719ba46ffc566e4ead/src/pages/ShareCodePage.js#L40




### What alternative solutions did you explore? (Optional)

**Comment by joekaufmanexpensify** [comment]
I can reproduce this, but I don't think this is a bug, and am not sure we should do anything to change this. Here is the language from the threads design doc discussing how we truncate the workspace name and root report. It looks like we applied this same logic to QR codes:

>Truncation of both the rows in the header will work the same. We’ll fit as much as available space will allow leaving room for “..” if either row exceeds that space.

cc @stitesExpensify Curious if you agree this is not a bug? It seems like we just did not optimize for workspace name longer than 30 characters when displaying both the workspace name and root report on a QR code (which is fair, since names this long shouldn't be common).

**Comment by stitesExpensify** [comment]
As in the bottom portion below the qr code would expand, and have multiple lines.

Some alternatives would be:
1. Do nothing
2. Switch the order so that the room name is always first, and then the workspace so that it is always visible. If we make them linkable though (like you can click to go up a layer) then it won't make as much sense

**Comment by joekaufmanexpensify** [comment]
[Brought question](https://expensify.slack.com/archives/C01SKUP7QR0/p1685049282950219?thread_ts=1685048738.217459&cid=C01SKUP7QR0) to Slack

**Comment by joekaufmanexpensify** [comment]
I see Stites is out of the office through the weekend. So going to add the external label to get more eyes on this.

**Comment by joekaufmanexpensify** [comment]
@rushatgabhane @dangrous we were discussing above what path forward to take here, including possibly doing nothing. I'm personally not sure we need to do anything, because of the reasoning I outlined [here](https://github.com/Expensify/App/issues/19359#issuecomment-1561083322). Curious what you both think though?

**Comment by joekaufmanexpensify** [comment]
I was covering this issue for @sophiepintoraetz, but now I'm OOO next week (until June 5th), so adding an additional assignee. Based on OOO, looks like Sophie will be back on 2023-05-31! 

Next steps here are to drive the discussion forward around whether we actually do want to do anything here. If so, there are a few proposals for C+ to review.

**Comment by alexpensify** [comment]
@dangrous - See Joe's notes here: https://github.com/Expensify/App/issues/19359#issuecomment-1564889640

We need some feedback on if we should continue forward.

**Comment by alexpensify** [comment]
No update

**Comment by dangrous** [comment]
If we're planning on limiting workspace names (I agree we should!) then as long as this is not an issue for the max possible length, I don't think we need to do anything. So:

**What is the planned max length for a workspace name (or a room, for that matter)?** It looks like we haven't decided that yet.

and, subsequently:

**Is this still an issue at those max lengths?**

If so, we should fix it as necessary. If not, let's close. So, for now, maybe we hold it for https://github.com/Expensify/App/issues/18647?

**Comment by alexpensify** [comment]
@sophiepintoraetz - I think you are back online. Let me know when I can unassign myself. Thanks!

**Comment by sophiepintoraetz** [comment]
Yup, thanks @joekaufmanexpensify and @alexpensify I'll take it from here! (or reassign yourselves if you want to maintain GH credit :D)

**Comment by dangrous** [comment]
Not overdue, on hold.

**Comment by sophiepintoraetz** [comment]
PR is on staging.

**Comment by dangrous** [comment]
Still waiting on #18647

**Comment by sophiepintoraetz** [comment]
@sonialiap - I'm headed OOO until 20 June, we're waiting on https://github.com/Expensify/App/issues/18647 to 1) see what the limit will be and 2) if this will still be an issue (we might increase the limit or something).

**Comment by sonialiap** [comment]
Still on hold

**Comment by sonialiap** [comment]
still on hold

**Comment by sophiepintoraetz** [comment]
Still on hold, taking this one back - thank you Sonia ❤️

**Comment by sophiepintoraetz** [comment]
Still holding.

**Comment by dangrous** [comment]
work on #18647 continues, looks like there's good momentum though!

**Comment by dangrous** [comment]
Same as above! Discussion continues on what exactly those limits should be. Once we're good with that, we can test and move forward with this one.

**Comment by dangrous** [comment]
Hi @joekaufmanexpensify! Can you confirm what we're moving forward with for workspace and room names? It sounds like from Slack that we're going to update it to 100 characters for both?

if so, we'll definitely need to keep this issue live and figure out how to adapt what we have - my guess is wrapping over line breaks is going to be the easiest thing here.

**Comment by joekaufmanexpensify** [comment]
Yep! We're looking at 100 for room names. So I agree we should fix the bug outlined in this issue.

**Comment by dangrous** [comment]
Okay great, I'm taking this issue off hold.

Updates:
* We're looking for solutions that will look clean with room name and workspace names up to 100 characters in length.

I'm wondering @shawnborton do you have time to make a rough mockup of how this might ideally look, and then we can take proposals on how to implement? We're looking specifically at the share QR codes such as this one. We need to allow for either a very long room name, a very long workspace name, or both.

https://user-images.githubusercontent.com/43996225/239775335-917061e5-3d88-43a9-85de-17e13022f87e.png

**Comment by dangrous** [comment]
On the thread chat, but not on the Share Code. That is what this issue is about, if I'm understanding it correctly (we should probably update the title so that's clear)

**Comment by rain2o** [comment]
As @dukenv0307 mentioned, I think there are some conflicting goals here. Not sure if this is a bug or a miscommunication across different goals and components.

The current version no longer shows the thread & workspace name in the QR view. It looks like [this PR](https://github.com/Expensify/App/pull/21092) updated it so that chat threads always return an empty string. Prior to that, [this PR](https://github.com/Expensify/App/pull/20484) changed the format of the subtitle (from `Workspace Name • #roomName` to `From #roomName in Workspace Name`. There have been several changes recently that seem to impact this subtitle.

We could bring back the subtitle for this QR component as part of this bug fix. But before doing that I think it's important that the final result is clear. Do we want to bring that back, was it removed unintentionally or are there conflicting goals at play? Once we have a clear destination, we can more accurately provide proposals for the path to said destination.

**Comment by dangrous** [comment]
Oh I see what you mean about the threads. However, it does look like there is still a lot of abbreviation for base workspace rooms. Do we want to adjust the design to show the full names, or are we fine with the `...`? cc @shawnborton thanks!

<img width="359" alt="image" src="https://github.com/Expensify/App/assets/5487802/8d366959-bb7a-4779-bc59-b98774335cbc">

**Comment by shawnborton** [comment]
Hmm maybe we allow the workspace line to go to 2 lines and then truncate?

**Comment by JmillsExpensify** [comment]
+1 to that.

**Comment by varshamb** [comment]
### Proposal
### Problem
In QR code page, the subtitle (eg: workspace name) is only single line and trucated.

### What is the root cause
We've set number of lines to 1.
https://github.com/Expensify/App/blob/0e2f74289b02912ff4eb2f0fafd7f8c9c389ace6/src/components/QRShare/index.js#L81

### What changes do I propose
Update the `numberOfLines` to 2

**Comment by rushatgabhane** [comment]
Looks like we only need to update the max lines to `2` and this issue will be resolved.
@dangrous let's assign @varshamb for their [proposal](https://github.com/Expensify/App/issues/19359#issuecomment-1650008253)?

**Comment by rushatgabhane** [comment]
> It seems we only allow the subtitle go to 2 lines if it is workspace name (not all subtitle)

i think all subtitle is fine. Can't think of any reason why we should do treat other subtitles differently

**Comment by dangrous** [comment]
so we're allowing up to 100 chars for each right now as per the work being done in https://github.com/Expensify/App/issues/18647

**Comment by shawnborton** [comment]
Cool, works for me. So I guess we don't worry about truncation here, and just center everything up.

**Comment by sophiepintoraetz** [comment]
Payouts due:

Issue Reporter: $250 @priya-zha
Contributor: $1500 @varshamb 
Contributor+: $1500 @rushatgabhane 

Eligible for 50% #urgency bonus? Y

**Comment by priya-zha** [comment]
Hey @sophiepintoraetz. Seems like you have sent me an offer for this issue now. Melvin had already sent me an offer earlier. I accepted it and I'm just waiting for you to send the payment and end the contract. You can withdraw the contract that you've sent me now.

**Comment by rushatgabhane** [comment]
@priya-zha FYI, you could also reject the offer on upwork

**Comment by rushatgabhane** [comment]
WIP ⚠️ 

1. The PR that introduced the bug has been identified. Link to the PR:

2. The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment:

3. A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion:

4. Determine if we should create a regression test for this bug.

5. If we decide to create a [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) for the bug, please propose the regression test steps to ensure the same bug will not reach production again

**Comment by rushatgabhane** [comment]
created a manual request - https://staging.new.expensify.com/r/3749561052279859

**Comment by sophiepintoraetz** [comment]
@priya-zha - please can you just accept the offer as it appears the original has timed out.

**Comment by sophiepintoraetz** [comment]
@varshamb - I believe I sent you an offer via Upwork, can you please confirm you have received it?|
https://www.upwork.com/jobs/~01b4b2a05ab4dd6216

**Comment by priya-zha** [comment]
> @priya-zha - please can you just accept the offer as it appears the original has timed out.

Okay. Accepted the offer that you sent me today. If the original has timed out, can you withdraw the offer that Melvin sent me earlier . I accepted that offer (melvin's) but the payment has not been done for that.

Thanks

**Comment by JmillsExpensify** [comment]
Reviewed details for @rushatgabhane. These details are accurate based on summary from Business Reviewer and are now approved for payment in NewDot.

**Comment by sophiepintoraetz** [comment]
Payments done here!
